spec/rspec/core/option_parser_spec.rb

```diff
@@ -317,6 +340,24 @@ def generate_help_text
       end
     end
 
+    describe "--bisect" do
+      it "sets the `:bisect` option" do
+        options = Parser.parse(%w[ --bisect ])
+
+        expect(options[:bisect]).to be_truthy
+      end
+
+      it "sets the `:runner` option with the `bisect` invocation" do
+        bisect = double(:bisect)
+        allow(RSpec::Core::Invocations).to receive(:bisect).and_return(bisect)
```

Comment:
The downside there is it doesn't let you capture extra info in the callable ([as required by `.print_help`](https://github.com/urbanautomaton/rspec-core/blob/support-bisect-in-spec-opts/lib/rspec/core/invocations.rb#L46-L52)). I do agree though, I don't particularly like the stubbing. Maybe we could turn them into callable objects?

```ruby
module Invocations
  PrintHelp = Struct.new(:parser, :invalid_options) do
    def call(options, err, out)
       #...
    end
  end
end
```

which would let us assert:

```ruby
expect(options[:runner]).to be_instance_of(RSpec::Core::Invocations::PrintHelp)
```